app/beacon-omop-worker/beacon_omop_worker/query_solvers.py

```diff
@@ -307,3 +309,167 @@ def solve_individuals(db_manager: SyncDBManager, query_terms: str):
+            logger.info(f"Successfully fetched condition data for concept_id {concept_id}.")
+            return data
+        except Exception as e:
+            logger.error(f"Error fetching condition data: {e}")
+            raise
+
+    def plot_kaplan_meier_curve(self, data: pd.DataFrame, concept_id: str, file_path: str) -> None:
+        """
+        Plot the Kaplan-Meier curve for a specific medical condition and save it to a file.
+        
+        Args:
+            data (pd.DataFrame): The data containing condition information.
+            concept_id (str): The concept_id for the condition.
+            file_path (str): The file path where the plot will be saved.
+        
+        """
+        logger = logging.getLogger(config.LOGGER_NAME)
+        try:
+            # Drop entries without start_date
+            data = data.dropna(subset=['condition_start_date'])
+
+            # Ensure dates are in correct format
+            data['condition_start_date'] = pd.to_datetime(data['condition_start_date'])
+            data['condition_end_date'] = pd.to_datetime(data['condition_end_date'])
+
+            # Handle censored data
+            data['observed'] = data['condition_end_date'].isna().apply(lambda x: 0 if x else 1)
+            censored_date = pd.Timestamp('2023-12-31')
+            data['condition_end_date'] = data['condition_end_date'].fillna(censored_date)
+
+            # Calculate duration
+            data['duration'] = ((data['condition_end_date'] - data['condition_start_date']).dt.days)
+
+            # Kaplan-Meier estimator fitting
+            event_times = data['duration']
+            event_observed = data['observed']
+            kmf = KaplanMeierFitter()
+            kmf.fit(event_times, event_observed)
+
+            # Plot the Kaplan-Meier curve
+            ax = kmf.plot(ci_show=True)
+            plt.xlabel('Time (days)')
+            plt.ylabel('Survival Probability')
+            plt.title(f'Kaplan-Meier Curve for Concept ID: {concept_id}')
+
+            # Customize ticks and add median survival line
+            plt.xlim(0, 15)
+            plt.xticks(ticks=range(0, 15, 1))
+            plt.yticks(ticks=[0, 0.2, 0.4, 0.6, 0.8, 1])
+            median_survival_time = kmf.median_survival_time_
+            plt.axvline(median_survival_time, color='r', linestyle='--', label=f'Median: {median_survival_time:.2f}')
+            plt.legend()
+
+            # Save the plot to the specified file
+            plt.savefig(file_path, format='png')  # You can change 'png' to 'pdf' if you prefer
+            plt.close()  # Close the plot to prevent display issues in non-GUI environments
+
+            logger.info(f"Kaplan-Meier plot saved to {file_path} for concept_id {concept_id}.")
+        except Exception as e:
+            logger.error(f"Error while plotting and saving Kaplan-Meier curve: {e}")
+            raise
```

Comment:
```suggestion
            raise e
```
Raise the exception that was thrown